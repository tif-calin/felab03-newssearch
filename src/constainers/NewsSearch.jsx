import React from 'react';
import PropTypes from 'prop-types';

const NewsSearch = props => {
  return <div className="NewsSearch">

  </div>;
};

export default NewsSearch;
