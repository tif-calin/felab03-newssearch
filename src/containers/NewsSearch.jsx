import React, { useEffect } from 'react';
import Search from '../components/search/Search';
import ArticleList from '../components/list/ArticleList';
import { fetchArticles } from '../services/apiNews.js';

const NewsSearch = () => {
  const [articles, setArticles] = React.useState([]);
  const [loading, setLoading] = React.useState(true);
  const [query, setQuery] = React.useState('florida');

  useEffect(async () => {
    setLoading(true);
    fetchArticles(query)
      .then(news => setArticles(news))
      .then(() => setLoading(false))
    ;
  }, [query]);

  return <div className="NewsSearch">
    <Search query={query} setQuery={setQuery}/>
    {loading ? <span>loading</span> : <ArticleList articles={articles}/>}
  </div>;
};

export default NewsSearch;
