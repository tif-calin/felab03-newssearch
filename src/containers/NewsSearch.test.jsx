import React from 'react';
import { render, cleanup } from '@testing-library/react';
import NewsSearch from './NewsSearch';

describe('NewsSearch component', () => {
  afterEach(() => cleanup());

  it('renders NewsSearch', () => {
    const { asFragment } = render(<NewsSearch />);
    expect(asFragment()).toMatchSnapshot();
  });
  
});
