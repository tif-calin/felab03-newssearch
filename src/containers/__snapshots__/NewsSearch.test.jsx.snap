// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsSearch component renders NewsSearch 1`] = `
<DocumentFragment>
  <div
    class="NewsSearch"
  >
    <div
      class="Search"
    >
      <input
        type="text"
        value="kahnawake"
      />
    </div>
    <span>
      loading
    </span>
  </div>
</DocumentFragment>
`;
